### PR TITLE
Entities/Creature: Spell facing, again, again, again.

### PR DESCRIPTION
**Pull request by Treeston**
Some clean-up, move code to places it really should've been to begin with.

Tagging issues #16572 and #17016.

### CODE REVIEW

**Comment by Killyana** [comment]
https://github.com/TrinityCore/TrinityCore/issues/17016 is fixed by this PR but not https://github.com/TrinityCore/TrinityCore/issues/16572

**Comment by Treeston** [comment]
I am aware, @Killyana. Haven't gotten to finish this up, need a clean way of dealing with idle orientation.

**Comment by Nawuko** [comment]
Whats wrong with using the "home" orientation?

**Comment by ccrs** [comment]
may be different, store orientation on cast?

File: src/server/game/Entities/Creature/Creature.h [hunk 1]
```diff
@@ -686,6 +686,7 @@ class TC_GAME_API Creature : public Unit, public GridObject<Creature>, public Ma
 
         // Handling caster facing during spellcast
         void SetTarget(ObjectGuid guid) override;
+        void MustReacquireTarget() { m_shouldReacquireTarget = true; } // flags the Unit for forced (client displayed) target reacquisition in the next ::Attack call
```
Comment (ccrs) [inline src/server/game/Entities/Creature/Creature.h]: copy paste error on comment

**Comment by Treeston** [comment]
There we go.

**Comment by ccrs** [comment]
what is missing here then? xd

**Comment by Treeston** [comment]
@ccrs nothing (test feedback from original reporter)

**Comment by ariel-** [comment]
I can vouch for this fixing correctly #17016 and #16572.

Good job there treeston :shipit:

**Comment by ccrs** [comment]
tested and working.

though there is an error on setfacing method related to movement.
as its called frequently on boss fights, the motion generator can mistake the ended facing spline with its own (the one it creates).
so edge cases can happen on encounters that involve any movement like LK
(it stays on position and calls movementInform because the movement spline is replaced by the facing one).

But like I said, its kinda unrelated and sorta out of scope for this PR.

**Comment by shenhuyong** [comment]
Tested and I found the drake vehicles in Oculus can't land now.

**Comment by ariel-** [comment]
This is correct, oculus drakes stop movement to player (presumably after an update tick) to change orient; copying the `IsStopped` check from `Unit::SetFacingToObject(WorldObject const*)` to `Unit::SetFacingTo(float)` seems to solve this issue. (I don't know if this introduces side effects)

```diff
void Unit::SetFacingTo(float ori)
{
+    // never face when already moving
+    if (!IsStopped())
+        return;

    Movement::MoveSplineInit init(this);
    init.MoveTo(GetPositionX(), GetPositionY(), GetPositionZMinusOffset(), false);
    if (HasUnitMovementFlag(MOVEMENTFLAG_ONTRANSPORT) && GetTransGUID())
        init.DisableTransportPathTransformations(); // It makes no sense to target global orientation
    init.SetFacing(ori);
    init.Launch();
}

**Comment by ariel-** [comment]
Another bug: Kologarn is turned around, due to vehicle accessory casting (should vehicle auras be affected by spell facing?)

`.go creature id 32930`

**Comment by Treeston** [comment]
No, they shouldn't. I'll probably just add a flat out disabling of facing for vehicle-related spells.

**Comment by ccrs** [comment]
wait I cannot reproduce that @ariel-

**Comment by Treeston** [comment]
So, I disabled it for vehicle auras, but Kologarn was still turned around.

I then spent an hour adding debug output to no avail.

Then I tried on latest 3.3.5. Kologarn is turned around there, too. This doesn't seem to be a spell facing issue (the arms don't cast anything).

@ariel-

**Comment by Shauren** [comment]
This ready now?

**Comment by Treeston** [comment]
Yes.

**Pull request merged**